File: apps/web/lib/feedback.ts
import { apiFetch } from "@/lib/api";

export interface SubmissionFeedback {
  id: string;
  recipient_id: string;
  coach_id?: string | null;
  grade?: number | null;
  feedback?: string | null;
  returned: boolean;
  created_at?: string | null;
  updated_at?: string | null;
}

export interface UpsertFeedbackRequest {
  grade?: number | null;
  feedback?: string | null;
  returned?: boolean | null;
}

export async function getFeedback(
  recipientId: string,
): Promise<SubmissionFeedback | null> {
  return apiFetch<SubmissionFeedback | null>(
    `/api/recipients/${recipientId}/feedback`,
  );
}

export async function upsertFeedback(
  recipientId: string,
  req: UpsertFeedbackRequest,
): Promise<SubmissionFeedback> {
  return apiFetch<SubmissionFeedback>(
    `/api/recipients/${recipientId}/feedback`,
    {
      method: "PUT",
      body: JSON.stringify(req),
    },
  );
}

File: apps/api/models/feedback.py
"""Models for coach submission feedback and manual grades."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class SubmissionFeedback(BaseModel):
    id: str
    recipient_id: str
    coach_id: str | None = None
    grade: float | None = None
    feedback: str | None = None
    returned: bool = False
    created_at: datetime | str | None = None
    updated_at: datetime | str | None = None


class UpsertFeedbackRequest(BaseModel):
    grade: float | None = None
    feedback: str | None = None
    returned: bool | None = None
